### PR TITLE
[timeseries] Preprocess real-valued features for neural network models

### PR DESCRIPTION
**Pull request by shchur**
*Description of changes:*
- Preprocess real-valued columns in known, past & static covariates for GluonTS models. We use the same logic as used by `TORCH_NN` model in AG-Tabular:
  - Boolean features are kept as-is
  - Skewed features (skewness > 0.99) are transformed with `QuantileTransform`
  - Remaining features are transformed with `StandardScaler`

To do:
- [x] Add tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by yinweisu** [comment]
| Previous CI Run | Current CI Run |
| :---: | :---: |

File: timeseries/src/autogluon/timeseries/models/gluonts/abstract_gluonts.py [hunk 1]
```diff
@@ -295,6 +298,69 @@ def _deferred_init_params_aux(self, dataset: TimeSeriesDataFrame) -> None:
...
+                self._real_column_transformers["static"] = self._get_transformer_for_columns(
+                    data.static_features, columns=columns
+                )
+            data.static_features[columns] = self._real_column_transformers["static"].transform(
+                data.static_features[columns]
+            )
+        return data
+
+    def _get_transformer_for_columns(self, df: pd.DataFrame, columns: List[str]) -> Dict[str, str]:
+        """Ignore bool features, use QuantileTransform for skewed features, and use StandardScaler for the rest.
```
Comment (canerturkmen) [inline timeseries/src/autogluon/timeseries/models/gluonts/abstract_gluonts.py:329]: nit: Ignore, i.e., they will "passthrough" right? Might help to make explicit.

**Review by canerturkmen — APPROVED** [review]
LGTM! Thanks!

**Comment by yinweisu** [comment]
| Previous CI Run | Current CI Run |
| :---: | :---: |

**Comment by yinweisu** [comment]
| Previous CI Run | Current CI Run |
| :---: | :---: |
| fonttools==4.50.0 | fonttools==4.51.0 |
| fonttools==4.50.0 | fonttools==4.51.0 |

**Pull request merged**